lightly/models/modules/masked_vision_transformer_timm.py

```diff
@@ -100,26 +137,55 @@ def encode(
                 If specified, only the indexed tokens will be encoded.
 
         Returns:
-            Batch of encoded output tokens.
+            Processed tokens for further processing.
+
         """
         # convert images to tokens
-        input = self.images_to_tokens(images)
+        tokens = self.images_to_tokens(images)
         # add prefix tokens if needed
-        input = self.add_prefix_tokens(input)
+        tokens = self.add_prefix_tokens(tokens)
 
         if idx_mask is not None:
-            input = utils.mask_at_index(input, idx_mask, self.mask_token)
+            tokens = utils.mask_at_index(tokens, idx_mask, self.mask_token)
         # add positional encoding
-        input = self.add_pos_embed(input)
+        tokens = self.add_pos_embed(tokens)
 
         if idx_keep is not None:
-            input = utils.get_at_index(input, idx_keep)
+            tokens = utils.get_at_index(tokens, idx_keep)
+
+        return tokens
+
+    def encode(
+        self,
+        images: Tensor,
+        idx_mask: Optional[Tensor] = None,
+        idx_keep: Optional[Tensor] = None,
+    ) -> Tensor:
+        """Encode input images.
+
+        Args:
+            images:
+                Batch of input images.
+            idx_mask:
+                Tensor with shape (batch_size, num_tokens_to_mask) where each
+                entry is an index of the token to mask in the respective batch.
+                If specified, the indexed tokens are masked with self.mask_token.
+            idx_keep:
+                Tensor with shape (batch_size, num_tokens_to_keep) where each
+                entry is an index of the token to keep in the respective batch.
+                If specified, only the indexed tokens will be encoded.
+
+        Returns:
+            Batch of encoded output tokens.
+        """
+        # preprocess images, convert to tokens and add positional embeddings
+        tokens = self.preprocess(images=images, idx_mask=idx_mask, idx_keep=idx_keep)
         # normalization layer
-        input = self.vit.norm_pre(input)
+        post_norm = self.vit.norm_pre(tokens)
```

Comment:
Feel free to call the variable `tokens` or `x` in the whole function. This keeps memory a bit smaller.